### ISSUE TITLE
[Sonic Forces] Controller not working with steam input

### ISSUE THREAD

**Issue by RookieFallGuy** [thesupersonic16/HedgeModManager#904]
## Describe the issue/bug
Whenever i turn on steam input my controller doesn't work however when i turn off the mod manager it suddenly works

## Expected Behaviour
I was expecting my controller to work with steam input and the mod manager together

## Additional Context
I've tried turning off steam input and it works but it has joystick and button problems like always moving to the right and double jumping without pressing the button twice

**Comment by thesupersonic16** [comment]
This issue may need to be transferred to [hedge-dev/HE2ModLoader](https://github.com/hedge-dev/HE2ModLoader). Will also mark this issue as bug so it will not close.

Some users have reported that enabling Steam Input causes Sonic Forces to fail with the mod loader enabled. This will need to be looked at.

**Comment by thesupersonic16** [comment]
Can you try renaming `dinput8.dll` to `d3d11.dll` and see if the modloader start working with Steam Input?

**Comment by RookieFallGuy** [comment]
> Can you try renaming `dinput8.dll` to `d3d11.dll` and see if the modloader start working with Steam Input?

Yes it works ty however it uninstalls he2 mod loader but the mods somehow work still?

**Comment by Carlnotaur** [comment]
I MANAGED TO FIX IT

 1. Go to %LOCALAPPDATA%l\NeverFinishAnything\CommonLoader\.storage

2. In the folder there will be several files that are strings of numbers with the .ini extension. Read them until you find one that starts with this:
[Application]
Name=Sonic Forces.exe





3. Delete that file, and reinstall the modloader if

**Comment by thesupersonic16** [comment]
> I MANAGED TO FIX IT
> 
> 1. Go to %LOCALAPPDATA%l\NeverFinishAnything\CommonLoader.storage
> 2. In the folder there will be several files that are strings of numbers with the .ini extension. Read them until you find one that starts with this:
>    [Application]
>    Name=Sonic Forces.exe
> 3. Delete that file, and reinstall the modloader if

@Carlnotaur Is your issue related to #653?

**Comment by thesupersonic16** [comment]
> > Can you try renaming `dinput8.dll` to `d3d11.dll` and see if the modloader start working with Steam Input?
> 
> Yes it works ty however it uninstalls he2 mod loader but the mods somehow work still?

@RookieFallGuy Sorry for the very late response. Want to be be sure, renaming the file to `d3d11.dll` fixed the controller issues while loading mods? And renaming it back broke it again?

HMM will think the modloader is not installed when renamed as most MLs are being transitioned to the new name to allow compatibility with some common graphic injectors. I am thinking this transition may have conflicted with Steam Input.

**Comment by RookieFallGuy** [comment]
> I MANAGED TO FIX IT
> 
> 1. Go to %LOCALAPPDATA%l\NeverFinishAnything\CommonLoader.storage
> 2. In the folder there will be several files that are strings of numbers with the .ini extension. Read them until you find one that starts with this:
>    [Application]
>    Name=Sonic Forces.exe
> 3. Delete that file, and reinstall the modloader if

**Comment by RookieFallGuy** [comment]
> I MANAGED TO FIX IT
> 
> 1. Go to %LOCALAPPDATA%l\NeverFinishAnything\CommonLoader.storage
> 2. In the folder there will be several files that are strings of numbers with the .ini extension. Read them until you find one that starts with this:
>    [Application]
>    Name=Sonic Forces.exe
> 3. Delete that file, and reinstall the modloader if

This didnt work sorry now it has lines of code of the mods i have along with the game

**Comment by RookieFallGuy** [comment]
> > > Can you try renaming `dinput8.dll` to `d3d11.dll` and see if the modloader start working with Steam Input?
> > 
> > 
> > Yes it works ty however it uninstalls he2 mod loader but the mods somehow work still?
> 
> @RookieFallGuy Sorry for the very late response. Want to be be sure, renaming the file to `d3d11.dll` fixed the controller issues while loading mods? And renaming it back broke it again?
> 
> HMM will think the modloader is not installed when renamed as most MLs are being transitioned to the new name to allow compatibility with some common graphic injectors. I am thinking this transition may have conflicted with Steam Input.

sorry if i took long but my controller works with the mods it just says i need to install he2 but if i say no it just opens the game and loads the mods normally with no issues
